src/services/Login.ts

```diff
@@ -0,0 +1,163 @@
+import * as API from './Api';
+
+import { AuthInfo, AuthStrategy } from '../types/Auth';
+import { Session } from 'src/store/Store';
+
+// We just mock the types for state for now, to avoid injecting more
+// dependencies on the service.
+type Dispatch = any;
+type AppState = any;
+
+// Stores the result of a computation:
+// hold = stop all computation and wait for a side-effect, such as a redirect
+// continue = continue...
+// success = authentication was a success, session is available
+// failure = authentication failed, session is undefined but error is available
+export enum Result {
```

Comment:
The **Result** is very generic there are many kinds of Results. So `AuthResult` or something like that to help comprehension.